catkin_tools/metadata.py

```diff
@@ -232,16 +232,6 @@ def init_metadata_root(workspace_path, reset=False):
             "Can't initialize Catkin workspace in path %s because it does "
             "not exist." % (workspace_path))
 
-    # Check if the desired workspace is enclosed in another workspace
-    marked_workspace = find_enclosing_workspaces(workspace_path)
-    if marked_workspace: marked_workspace = marked_workspace[0]
-
-    if marked_workspace and marked_workspace != workspace_path:
-        raise IOError(
-            "Can't initialize Catkin workspace in path %s because it is "
-            "already contained in another workspace: %s." %
-            (workspace_path, marked_workspace))
```

Comment:
Why removing this part? It does not seem related to the same issue.

Reply:
That code part is triggered by `catkin init` and it checks whether a workspace is enclosed in another workspace. If yes, `catkin` refuses to initialize a new workspace, even if you ask for it explicitly with `catkin init -w path/to/sub/workspace`.

In our patched version (with https://github.com/Intermodalics/catkin_tools/pull/3) that is fine, and I sometimes use this for tests in a temporary nested workspace, for example for a single package or for a submodule. But by default `catkin` will always find and use the enclosing workspace, so you have to pass `-w` explicitly when referring to the nested workspace.

But we could print a warning only, instead of removing the check completely...